### PR TITLE
refactor(core,console): refactor AC 403 flow

### PR DESCRIPTION
**Pull request by simeng-li**
## Testing

 refactor AC 403 flow

- Add global confirm modal provider 
```js
const {show} = useConfirmModal();

await show({ModalContent});
```

- Add global 403 API error handler, show confirm modal
- Auto signout and redirect user to the sign-in page after click on the confirm button

image

## Checklist

- [x] `.changset-staged`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [ ] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/console/src/hooks/use-api.ts [hunk 1]
```diff
@@ -56,8 +73,8 @@ export const useStaticApi = ({
           beforeError: hideErrorToast
             ? []
             : [
-                (error) => {
-                  void toastError(error.response);
+                async (error) => {
+                  await toastError(error.response);
```
Comment (charIeszhao) [inline packages/console/src/hooks/use-api.ts:77]: Why do we `await` here?

File: .changeset-staged/eight-rocks-wave.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+"@logto/console": patch
+"@logto/core": patch
+---
+
+## Refactor the Admin Console 403 flow
+
+- Add 403 error handler for all AC API requests
+- Show confirm modal to notify the user in not authorized
```
Comment (charIeszhao) [inline .changeset-staged/eight-rocks-wave.md:9]:
```suggestion
- Show confirm modal to notify the user who is not authorized
```

File: .changeset-staged/eight-rocks-wave.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+- sign out and redirect user to the sign-in page
```
Comment (charIeszhao) [inline .changeset-staged/eight-rocks-wave.md:10]:
```suggestion
- Click `confirm` button to sign out and redirect user to the sign-in page
```

File: packages/console/src/hooks/use-api.ts (hunk 1 above)
Comment (simeng-li) [reply packages/console/src/hooks/use-api.ts:77]: Hold page render till confirm modal is addressed

**Review by darcyYe — APPROVED** [review]

**Pull request merged**